### DISCUSSION TITLE
caddy su macchina dietro a vpn

### DISCUSSION THREAD

**Discussion by ster1um** [indomus/forum#899] [Networking]
Ho configurato caddy per rispondere sulla porta 9000 come da guida.
Sul server remoto dove gira la vpn ho fatto un prerouting dalla porta 443 > 443 e della porta 9000>9000  
quando cerco di raggiungere con il dominio liscio ricevo correttamente il respond se aggiungo la porta 9000 al dominio ricevo un BAD REQUEST.
Credo quindi che il problema sia lato home assistant

Sia Home Assistant core che caddy sono su docker

**Allego caddyfile**
```
(https_header) {
  header {
    Strict-Transport-Security "max-age=31536000; includeSubdomains"
    X-XSS-Protection "1; mode=block"
    X-Content-Type-Options "nosniff"
    X-Frame-Options "SAMEORIGIN"
    Referrer-Policy "same-origin"
  }
}
https://xxxxx.com { 
  respond "Saluti dal Crew inDomus"
  file_server
}
https://xxxxx.com:9000 { 
  import https_header
  reverse_proxy http://192.168.xx.xxx:8123
}
```

**configuartion.yaml**
```
http:
  use_x_forwarded_for: true
  trusted_proxies: 
    - 127.0.0.1
```

**Comment by snakuzzo** [comment]
Per favore correggi il post ed usa il markdown per i blocchi di codice

**Reply by snakuzzo** [reply]
Poi non hai spiegato come hai tirato su tutto quanto...come container docker ? come servizi ?

**Reply by ster1um** [reply]
> 
> 
> Poi non hai spiegato come hai tirato su tutto quanto...come container docker ? come servizi ?

scusa, aggiornato il post.

**Reply by snakuzzo** [reply]
Innanzi tutto, se hai creato il container di Caddy con network mode bridge (quello di default) mappando le singole porte, devi assicurarti di mappare anche la 9000.
Poi, il blocco `http:` della configurazione di Home Assistant, è sbagliato.
Nei `trusted_proxies` ci va l'IP da cui arrivano le chiamate verso HA dall'esterno, che non potrà mai essere `127.0.0.1` (visto che sarebbe la loopback dello stesso container di HA).
Lì devi metterci l'ip di Caddy o comunque la subnet docker con la quale Caddy contatta HA.
Se provi un accesso dall'esterno ora non funziona, ma ti basta andare a vedere nei log di HA da quale IP hai ricevuto una chiamata non valida...vedrai che quello sarà l'IP del container di Caddy e sarà di classe 172.

**Reply by ster1um** [reply]
ci provo e ti faccio sapere!

Grazie per il momento

agg.
come dicevi tu andava inserita l'ip del container di caddy.
per il resto avevo fatto tutto giusto anche la porta al docker di caddy.

**Reply by snakuzzo** [reply]
Perfetto ;)

**Reply by ster1um** [reply]
Ultima domanda non è possibile uscire su internet senza l'utilizzo della porta?

**Reply by snakuzzo** [reply]
Che intendi dire ?

**Reply by ster1um** [reply]
Intendo che ora per chiamare HA attraverso internet dovrò usare dominio.xxx:9000 (o la porta che desidero) ma non potrò più usare il dominio liscio...
Se faccio la chiamata al dominio senza porta mi risponde lo script "Saluti dal Crew inDomus"

**Reply by Scialla** [reply]
devi fare un reverse proxy dal tuo dominio "liscio" al tuo ip interno:8123

**Reply by snakuzzo** [reply]
Come dice anche @Scialla, ti basta sostituire quello che hai con questo...

```
(https_header) {
  header {
    Strict-Transport-Security "max-age=31536000; includeSubdomains"
    X-XSS-Protection "1; mode=block"
    X-Content-Type-Options "nosniff"
    X-Frame-Options "SAMEORIGIN"
    Referrer-Policy "same-origin"
  }
}

https://xxxxx.com { 
  import https_header
  reverse_proxy http://192.168.xx.xxx:8123
}
```

**Reply by ster1um** [reply]
> 
> 
> Come dice anche @Scialla, ti basta sostituire quello che hai con questo...
> 
> ```
> (https_header) {
>   header {
>     Strict-Transport-Security "max-age=31536000; includeSubdomains"
>     X-XSS-Protection "1; mode=block"
>     X-Content-Type-Options "nosniff"
>     X-Frame-Options "SAMEORIGIN"
>     Referrer-Policy "same-origin"
>   }
> }
> 
> https://xxxxx.com { 
>   import https_header
>   reverse_proxy http://192.168.xx.xxx:8123
> }
> ```
si si certo ci ero già arrivato da solo dopo l'intervento di scialla

Grazie mille ad entrambi
